### ISSUE TITLE
Asterisk afkeuren 1.1.1 op basis dat het niet door alle screenreaders (TalkBack) goed wordt voorgelezen

### ISSUE THREAD

**Issue by RenateRoke** [WCAG-Audit-Discussions/NL-BE#55]
Hoi allemaal, ik zag al een discussie die hier een beetje op lijkt maar me net niet het antwoord geeft waar wij naar op zoek zijn:

Vooralsnog keuren wij het alléén gebruiken van een asterisk (*) om aan te geven dat een veld verplicht is af, omdat TalkBack deze niet voorleest (als er geen spatie tussen het woord en de asterisk zit). Dit is nog steeds een openstaand known issue: https://issuetracker.google.com/issues/271387526. Ons advies als oplossing is dan altijd om aria-required op het invoerelement toe te voegen. 

Voor mij klinkt het logisch dat dit afkeurbaar is als we testen voor 'gangbare browsers en hulpsoftware', maar de meningen zijn nogal verdeeld. Hoe gaan andere bureaus/testers hier mee om en (vooral benieuwd naar) waarom? 

Edit: en op 1.1.1 omdat we een asterisk in z'n eentje interpreteren als non-text (en daar waren de meningen ook al over verdeeld zag ik haha).

Related: https://github.com/WCAG-Audit-Discussions/NL-BE/issues/40

**Comment by bramd** [comment]
Bij Firm Ground keurden we het niet af, maar ik sta ervoor open om dat te herzien. Ik ben benieuwd naar andere reacties.

**Comment by Veyfeyken** [comment]
Ik denk niet dat een bug in een bepaald hulpmiddel een doorslaggevend argument mag zijn in het beoordelen van een succes criterium. Als de betekenis van de asterisk verklaard wordt, bij voorkeur boven het formulier, keur ik het niet af.

1.1.1 Non-text content is in mijn mening niet van toepassing. Het is een karakter/leesteken, geen afbeelding maar ik begrijp wel dat daar de meningen verdeeld zijn.

> 3.3.2: Labels or Instructions
> Labels or instructions are provided when content requires user input.

Wanneer de asterisk verklaard wordt, vind ik dat aan bovenstaande is voldaan. 
Ik vermeld wel altijd dat het gebruik van een asterisk geen goed patroon is, tekstuele labels zijn de best practice.

**Comment by Aircl0wn** [comment]
Zoals de andere discussie al liet zien zijn de meningen hier nogal verdeeld vooralsnog.

Net als @Veyfeyken zie ik een * niet als non-text. Naast dat het een eigen ASCII symbool is, is het onderdeel van de `sequence of characters` die het label opmaken en die iets in menselijke taal probeert duidelijk te maken. 1.1.1 valt voor mij dan dus af.

Ik deel ook de mening dat een bug in een tool geen reden is tot afkeuren als het elders wel werkt.

Wat andere SC langlopend:

- 2.4.6: Als het sterretje niet verklaard wordt, is het label niet duidelijk genoeg.
- 3.3.2: Er een label aanwezig, maar of die duidelijk genoeg is, wordt in dit SC niets over gezegd. Er staat geen label AND instructions (helaas).
- 1.3.1: de bedoeling van het sterretje is een verplicht veld, dus moet je dit ook programmatische duidelijk maken. Dat kan natuurlijk op verschillende manieren, maar ik vind een sterretjes, zelfs uitgelegd, geen programmatische bepaling of vallen onder 'available in text' als dit niet letterlijk in het label verwerkt is.

**Comment by rvantonisse** [comment]
> 1.3.1: de bedoeling van het sterretje is een verplicht veld, dus moet je dit ook programmatische duidelijk maken. Dat kan natuurlijk op verschillende manieren, maar ik vind een sterretjes, zelfs uitgelegd, geen programmatische bepaling of vallen onder 'available in text' als dit niet letterlijk in het label verwerkt is.- @Aircl0wn 

Volgens mij geldt dit onder 4.1.2 voor invoervelden. Bij 1.3.1 is een tekstuele verwijzing voldoende wat wordt gedaan met de referentie / voetnoot stijl.

Met een situatie als deze:
```
<label for="name">Name*</label>
<input type="text" id="name">

<p>...content...</p>

<p>*: Required</p>
```

Vervelend dat het niet voorgelezen wordt, want dan is het ook onduidelijk. Ik zou er een opmerking over plaatsen aangezien het om 1 toepassing gaat, Talkback. Tenzij natuurlijk essentieel voor de doelgroep van de website en ondersteuning voor Talkback nodig is.

Terugkomend op 4.1.2, moet deze informatie technisch aanwezig zijn voor interface componenten. En dat zou ik met html `required` attribuut aangeven. Talkback lijkt op dit gebied het slechtst te scoren (https://a11ysupport.io/?featuresearch=required), dus des te belangrijker dat deze missende spaties eruit worden gewerkt voor de gebruikers van Talkback.

**Comment by Aircl0wn** [comment]
Ik blijf toch bij 1.3.1 want het is een visuele relatie (sterretje) die je programmatisch moet onderscheiden. Ik ben het eens dat een duidelijk label ook voldoende zou zijn. De vraag is of een sterretje, dat zelf weer elders wordt verklaard, duidelijk genoeg is. ik zelf zou zeggen van niet. je vraagt wel veel van iemand die die verklaring niet direct kan zien. Als het sterretje lokaal een verborgen tekst variant heeft, zou ik er geen probleem mee hebben, want dan heb je inderdaad die lokale verklaring.

Qua 4.1.2:
`required` is een state. 
de `name` en `role' moeten programmatisch aanwezig zijn. 
De `state` moet programmatisch 'instelbaar' zijn (gezet kunnen worden) _als_ de gebruiker dit kan instellen/aanpassen. Dat laatste is niet van toepassing hier want je bepaald het als auteur dat een veld verplicht is. dus mijns inziens gaat deze niet op hier.

Dit in tegenstelling tot een uitklapelement met een `aria-expanded` state bijvoorbeeld waarbij je de state bijhoudt als de gebruiker de interactie aangaat. Datzelfde attribuut is dus voor 1.3.1 ook belangrijk, want dit communiceert ook de mogelijke interactie.

Het gebruik van `required` moet je wel je validatie uit zetten, anders zit je met ontoegankelijke validatie meldingen. 
a11ysupport is vaak erg out-of-date helaas, dus daar kun je ook niet 1-2-3 van uit gaan.

**Comment by JJdeGroot** [comment]
Als de `required` state niet duidelijk is bij gebruik van een `*` zou ik het op 1.3.1 of 2.4.6 afkeuren. 

Als de asterisk uitgelegd wordt voordat de formulier velden beginnen, is het wat mij betreft duidelijk.
En een asterisk in combinatie met `aria-required` of vergelijkbaar is naar mijn idee ook duidelijk.

Dat een bepaalde hulpsoftware zoals TalkBack niet goed om gaat met bepaalde karakters is naar mijn idee geen afkeurpunt.

**Comment by RenateRoke** [comment]
Aangezien in het understanding document van 1.3.1 specifiek wordt benoemd dat het uitleggen van een asterisk als aanduiding van een verplicht veld voldoende is, gaan we er bij dezen niet meer vanuit dat het dan nog afkeurbaar is onder 1.1.1 maar we blijven het meenemen als advies :) dank allemaal!

**Comment by erikkroes** [comment]
Dus zonder uitleg afkeuren, en met uitleg meenemen als advies?

https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#examples

**Comment by rianrietveld** [comment]
Is de conclusie en de concensus dat een asterisk alleen, als visuele aanduiding voor een foutmelding:
- wordt beschouwd als non-text content (SC 1.1.1 Niet-tekstuele content)
- onvoldoende uitleg is (SC 3.3.2 Labels of instructies), er moet een uitleg van wat de asterisk betekent boven het formulier staan.

Geef een 👍 voor mee eens.
Gef een 👎 voor niet mee eens en vertel waarom.

Het besluit in dit issue geldt ook voor #40 (Is een asterisk zonder verklaring bij verplichte velden een failure? )

**Comment by janitatop** [comment]
Het lijkt mij niet toepasselijk onder 1.1.1 omdat ik het wel zie als tekst. Ik ben het er wel mee eens dat het teken verklaard moet worden bovenaan het formulier (punt 2). En voor hulpsoftware zou ik adviseren om aria-required toe te voegen voor als de asteriks niet goed wordt voorgelezen.

**Comment by cstrobbe** [comment]
Volledig oneens met afkeuren onder SC.1.1.1. Het TalkBack-probleem leidt tot een onvoldoende duidelijk label, wat onder SC ~~3.3.2~~ *2.4.6* valt.
(Een verklaring voor de asterisk boven het formulier is geen voorbeeld van SC 1.1.1 omdat die verklaring niet aan de definitie van [text alternative](https://www.w3.org/TR/WCAG21/#dfn-text-alternative) voldoet. Een "text alternative" is "programmatically associated with non-text content or referred to from text that is programmatically associated with non-text content".)

**Comment by Aircl0wn** [comment]
Ik delel de mening om het niet af te keuren onder SC 1.1.1.

Het niet ondersteunen door TalkBack zou ik alleen niet aanduiden als een failure voor SC 3.3.2. 
Als we dit voor alle 'bugs' in software moeten gaan doen is het einde zoek. Als de implementatie goed is en het zou in principe moeten werken keur ik het niet af.

**Comment by rianrietveld** [comment]
We laten afkeuren voor SC 1.1.1 vallen. Blijft 3.3.2 staan. 

Is de conclusie en de concensus dat een asterisk alleen onvoldoende uitleg is (SC 3.3.2 Labels of instructies), er moet een uitleg van wat de asterisk betekent boven het formulier staan.

Geef een 👍 voor mee eens.
Geef een 👎 voor niet mee eens en vertel waarom.

Het besluit in dit issue geldt ook voor https://github.com/WCAG-Audit-Discussions/NL-BE/issues/40 (Is een asterisk zonder verklaring bij verplichte velden een failure? )

**Comment by Veyfeyken** [comment]
Aan 3.3.2 Labels of instructies is voldaan want er is een label. Het gaat om 2.4.6 Koppen en labels. 

> Headings and [labels](https://www.w3.org/TR/WCAG22/#dfn-labels) describe topic or purpose.

Je kan argumenteren dat hier niet aan is voldaan als de asterisk niet is verklaard. Persoonlijk vind ik het geen failure. De asterisk verklaren lijkt me een best practice. Het niet verklaren vormt geen specifiek probleem voor mensen met een beperking, het is eerder een algemeen issue.

**Comment by Aircl0wn** [comment]
@Veyfeyken Klopt Gijs, is was in de war! (niet de eerste keer bij deze twee...)

**Comment by rianrietveld** [comment]
@RenateRoke we sluiten geen issues, we markeren ze met een label Concensus of Consensus onduidelijk". Anders verdwijnt de info. Als er consensus is, schrijf je dit als laatste comment en verander je het label/of laat je dit veranderen.

Is er consencus?

**Comment by rianrietveld** [comment]
Als er consensus is, schrijf je dit als laatste comment en verander je het label.

**Comment by cstrobbe** [comment]
> Ik blijf toch bij 1.3.1 want het is een visuele relatie (sterretje) die je programmatisch moet onderscheiden. ([Aircl0wn](https://github.com/Aircl0wn))

De aanwezigheid van een sterretje in een label is geen aspect van presentatie in de zin bedoeld door SC 1.3.1 (in tegenstelling tot positionering van een visueel label in de onmiddellijke nabijheid van een veld); het is informatie die beschikbaar is in tekst ("available in text").

Het sterretje is inhoud van het label; de vraag of het label voldoende beschrijvend is valt onder SC 2.4.6.

> Volgens mij geldt dit onder 4.1.2 voor invoervelden. ([rvantonisse](https://github.com/rvantonisse))

SC 4.1.2 werd eerder met het oog op custom UI-componenten e.d. geschreven. En trouwens, als het sterretje in het label staat (SC 2.4.6) en het label correct verbonden is met het veld (SC 1.3.1 voor iets waar de visuele relatie een label suggereert), dan heeft het veld al een toegankelijke naam in de zin van SC 4.1.2.

Tenslotte een vraag aan @RenateRoke : is dit ongeveer anderhalf jaar nadat dit ticket geopend werd nog steeds een bug in de geteste versie van Android?

**Comment by RenateRoke** [comment]
@cstrobbe Het issue is gesloten en ik zie sindsdien geen nieuwe issues meer die hieraan zijn gerelateerd en recenter zijn. In versie 14 werkt het, ik heb helaas geen oudere android versie beschikbaar om dit zelf te controleren.

Het oorspronkelijke issue is daarmee in ieder geval opgelost, maar er blijft alsnog een interessante discussie staan.

Ik heb me de afgelopen dagen rot gezocht naar iéts in normatieve teksten om het visuele label * zonder uitleg echt ergens af te keuren, of dat nou bij 1.3.1, 2.4.6 of 3.3.2 is. Ik kan eigenlijk niet echt tot een andere conclusie komen dat het een best practice is om het te verklaren en dat dit meer in het werkveld van onze ux-vrinden thuishoort om hier kracht achter te zetten.

Ik blijf hoop houden dat er wel iets boven water komt, dus gooi het op me als je het hebt gevonden, maar ik heb het niet kunnen vinden.
